NeoML/include/NeoML/TraditionalML/FloatVector.h

```diff
@@ -98,26 +100,52 @@ class NEOML_API CFloatVector {
 private:
 	// The body of the vector is an object containing all its data.
 	struct NEOML_API CFloatVectorBody: public IObject {
+		CSparseFloatVectorDesc Desc;
 		CFastArray<float, 1> Values;
 
-		explicit CFloatVectorBody( int size ) { Values.SetSize( size ); }
+		explicit CFloatVectorBody( int size );
 	
 		CFloatVectorBody* Duplicate() const;
 	};
 
 	CCopyOnWritePtr<CFloatVectorBody> body; // the vector body
+	CSparseFloatVectorDesc desc;
 };
 
 inline CFloatVector::CFloatVectorBody* CFloatVector::CFloatVectorBody::Duplicate() const
 {
 	auto result = FINE_DEBUG_NEW CFloatVectorBody( Values.Size() );
 	Values.CopyTo( result->Values );
+	result->Desc.Values = result->Values.GetPtr();
```

Comment:
You already did it in the constructor!

Reply:
agreed. I thought CopyTo() may reallocate the buffer, but here it seems ok.